### DISCUSSION TITLE
Hybrid dynamics

### DISCUSSION THREAD

**Discussion by shamilmamedov** [stack-of-tasks/pinocchio#1864] [Q&A] [answered]
Does pinocchio support hybrid dynamics? If it doesn't, is there a simple way to implement it on top of pinocchio? To give a bit of context, I have a serial chain floating base robot. I want to assign acceleration to the floating base joint and compute the accelerations of other joints using forward dynamics.

**Comment by jcarpent** [comment, marked as answer]
Good question. You should be able to do that with the constrained dynamics functions by setting the desired acceleration of the floating base. 
In general, it would be nice to add the feature overall.
In which context do you use this feature?

**Reply by shamilmamedov** [reply]
@jcarpent, thank you for quick reply!  Is there an example or unittest that I can get inspiration from?

I am using this feature in the context of deformable object manipulation where an object is attached to the end-effector of the robot. For time being I assume that I can directly assign the end-effector acceleration, thus need to solve hybrid dynamics problem.

**Reply by jcarpent** [reply]
You can take inspiration from https://github.com/stack-of-tasks/pinocchio/blob/c7530c9731095f6f3e5164ac8e4b925987079f39/unittest/contact-dynamics.cpp#L303-L360

**Reply by shamilmamedov** [reply]
@jcarpent thank you for the reference!

**Comment by shamilmamedov** [comment]
For those of you who are interested in computing hybrid dynamics using Pinocchio, I have discovered a method to do so using inverse dynamics implementation. The details of this method are explained in Roy Featherstone's book. Additionally, I have written a short [blog post](https://shamilmamedov.com/blog/2023/robot-dynamics/) where I briefly outline the method and provide a simple code snippet demonstrating how to implement it in Pinocchio.

**Reply by jcarpent** [reply]
Very nice initiative @shamilmamedov. Your blog post might be very helpful for people.

We may already think to implement such a solution in C++ directly for efficiency reasons.
@shamilmamedov Would you like to be involved in this task?

**Reply by shamilmamedov** [reply]
@jcarpent, I would like to, however, unfortunately, I am not an expert in C++ nor in spatial algebra
